File: Medium/45.Jump Game II/solution.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        for i in range(1, len(nums)):
            nums[i] = max(nums[i] + i, nums[i-1])

        ind = 0 
        ans = 0
        while(ind < len(nums) - 1):
            ans += 1
            ind = nums[ind]
        
        return ans

File: Medium/45.Jump Game II/solution.java
class Solution {
    public int jump(int[] nums) {
    for(int i = 1; i < nums.length; i++)
      {
        nums[i] = Math.max(nums[i] + i, nums[i-1]);
      }

      int ind = 0;
      int ans = 0;

      while(ind < nums.length - 1)
      {
        ans++;
        ind = nums[ind];
      }

      return ans;
    }
}
